qir/qsharp-tests/qsharp_tests/__init__.py

```diff
@@ -0,0 +1,139 @@
+
+import os
+import glob
+import shutil
+import tempfile
+import xml.etree.ElementTree as ET
+import subprocess
+import pytest
+
+BASE_DIR = os.path.dirname(__file__)
+QSHARP_SAMPLES = os.environ.get("QSHARP_SAMPLES")
+
+
+class QsharpProject(object):
+    def __init__(self, project_dir, filename):
+        self.original_project_dir = project_dir
+        self.filename = filename
+        self.has_relative_dependencies = False
+        self.skip = False
+
+    def __enter__(self):
+        self.project_dir = tempfile.mkdtemp()
+
+        # Copying project
+        shutil.copytree(self.original_project_dir, self.project_dir, dirs_exist_ok=True)
+        self.project_file = os.path.join(self.project_dir, self.filename)
+
+        if not os.path.isfile(self.project_file):
+            raise BaseException("{} does not exist".format(self.project_file))
+
+        # Updating project to emitting QIR
+        tree = ET.parse(self.project_file)
+        root = tree.getroot()
+
+        # Updating to QIR generation
+        for group in root.iter('PropertyGroup'):
+            qir_generation = group.find("QirGeneration")
+            if qir_generation is None:
+                qir_generation = ET.SubElement(group, 'QirGeneration')
+            qir_generation.text = "true"
+
+            library_type = group.find("OutputType")
+            if library_type is not None:
+                library_type.text = "Exe"
+
+        for group in root.iter('ItemGroup'):
+            if group.find("ProjectReference") is not None:
+                self.has_relative_dependencies = True
```

Comment:
In theory, QIR generation for projects with relative dependencies should work, but I agree that scoping it this way for now is a good way to focus on a subset of samples for the time being (and a neat mechanism too).

Reply:
The reason why I have to skip the ones with relative dependence is that I copy each project out to a temporary folder before modifying it such that you can use it with your local setup without destroying the repo. In principle, you could fix the relative dependence by referring back to the original repo, but it seems like a lot of work and very little benefit.  I am happy to review this at a later point though.